presto-hbase/src/main/java/com/facebook/presto/hbase/HBaseInternalFieldDescription.java

```diff
@@ -0,0 +1,281 @@
+
+    /**
+     * <tt>_key_length</tt> - length in bytes of the key.
+     */
+    public static final HBaseInternalFieldDescription KEY_LENGTH_FIELD = new HBaseInternalFieldDescription("_key_length", BigintType.BIGINT, "Total number of key bytes");
+
+    public static Set<HBaseInternalFieldDescription> getInternalFields()
+    {
+        return ImmutableSet.of(PARTITION_ID_FIELD, PARTITION_OFFSET_FIELD,
+                SEGMENT_START_FIELD, SEGMENT_END_FIELD, SEGMENT_COUNT_FIELD,
+                KEY_FIELD, KEY_CORRUPT_FIELD, KEY_LENGTH_FIELD,
+                MESSAGE_FIELD, MESSAGE_CORRUPT_FIELD, MESSAGE_LENGTH_FIELD);
+    }
+
+    private final String name;
+    private final Type type;
+    private final String comment;
+
+    HBaseInternalFieldDescription(
+            String name,
+            Type type,
+            String comment)
+    {
+        checkArgument(!isNullOrEmpty(name), "name is null or is empty");
+        this.name = name;
+        this.type = requireNonNull(type, "type is null");
+        this.comment = requireNonNull(comment, "comment is null");
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public Type getType()
+    {
+        return type;
+    }
+
+    HBaseColumnHandle getColumnHandle(String connectorId, int index, boolean hidden)
+    {
+        return new HBaseColumnHandle(connectorId,
+                index,
+                getName(),
+                getType(),
+                null,
+                null,
+                null,
+                false,
+                hidden,
+                true);
+    }
+
+    ColumnMetadata getColumnMetadata(boolean hidden)
+    {
+        return new ColumnMetadata(name, type, comment, hidden);
+    }
+
+    public FieldValueProvider forBooleanValue(boolean value)
+    {
+        return new BooleanHBaseFieldValueProvider(value);
+    }
+
+    public FieldValueProvider forLongValue(long value)
+    {
+        return new LongHBaseFieldValueProvider(value);
+    }
+
+    public FieldValueProvider forByteValue(byte[] value)
+    {
+        return new BytesHBaseFieldValueProvider(value);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(name, type);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        HBaseInternalFieldDescription other = (HBaseInternalFieldDescription) obj;
+        return Objects.equals(this.name, other.name) &&
+                Objects.equals(this.type, other.type);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("name", name)
+                .add("type", type)
+                .toString();
+    }
+
+    public class BooleanHBaseFieldValueProvider
```

Comment:
I would make these inner methods `static` and pass `name` explicitly to instances via constructor.